### PR TITLE
Fixed stripe INCREMENTAL run and updated tap

### PR DESCRIPTION
**Pull request by Luishfs**
# Description

Fixed stripe INCREMENTAL run, where the tap would add a sufix to certain bookmarks and break INCREMENTAL runs
Updated Tap to newer version and stripe python lib to version 5.5.0

# How Has This Been Tested?

Tests were executed using a demo Stripe account with PostgreSQL as a destination

## FULL TABLE run
Screenshot from 2023-10-24 17-12-43

## INCREMENTAL run
Screenshot from 2023-10-24 17-12-50
Screenshot from 2023-10-24 17-12-57

## Schema Example
Screenshot from 2023-10-24 16-48-41

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [X] I have added unit tests that prove my fix is effective or that my feature works
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_integrations/mage_integrations/sources/stripe/__init__.py [hunk 1]
```diff
@@ -205,11 +207,7 @@ def get_schema(cls, stream_name):
 
     @classmethod
     def is_selected(cls, stream_name):
-        # TODO (tommy dang): support syncing multiple streams aka sub-streams
         stream = cls.get_catalog_entry(stream_name)
-        if not stream:
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/stripe/__init__.py:210]: why do you remove this?

**Comment by wangxiaoyou1993** [comment]
Could you point me to the code that fixes the incremental run?
There're many other code changes in this PR.

**Comment by wangxiaoyou1993** [comment]
why do you update the file permission? 
image

File: mage_integrations/mage_integrations/sources/stripe/__init__.py [hunk 2]
```diff
@@ -368,10 +366,11 @@ def get_discovery_metadata(schema, key_properties, replication_method, replicati
     return metadata.to_list(mdata)
 
 
-def discover() -> Dict:
+def discover(logger=LOGGER):
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/stripe/__init__.py:369]: Don't set an mutable object as the default value.
Do
```
df discover(logger=None):
    if logger is None:
        logger = LOGGER
```

File: mage_integrations/mage_integrations/sources/stripe/__init__.py [hunk 3]
```diff
@@ -462,6 +461,27 @@ def reduce_foreign_keys(rec, stream_name):
...
+        method.lower(), url, params, headers, is_streaming=False
+    )
+    resp = self.interpret_response(rbody, rcode, rheaders)
+    LOGGER.debug('request id : %s', resp.request_id)
+    return resp, my_api_key
+
+
+# To log the request_id, we replaced the request() function of the APIRequestor
+# class o SDK, captured the response and logged the request_id
+APIRequestor.request = new_request
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/stripe/__init__.py:482]: What's this for?

File: mage_integrations/mage_integrations/sources/stripe/__init__.py [hunk 4]
```diff
@@ -952,21 +965,22 @@ def recursive_to_dict(some_obj):
 
 
 def sync_event_updates(stream_name, is_sub_stream):
-    """
+    '''
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/stripe/__init__.py:968]: Why do you change it to three single quote here?

File: mage_integrations/mage_integrations/sources/stripe/__init__.py [hunk 5]
```diff
@@ -1246,5 +1312,5 @@ def new_request(self, method, url, params=None, headers=None):
             Context.print_counts()
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/stripe/__init__.py:1315]: Why do you update it to double quotes here?

File: mage_integrations/mage_integrations/sources/stripe/__init__.py [hunk 6]
```diff
@@ -1121,20 +1136,71 @@ def write_bookmark_for_event_updates(is_sub_stream, stream_name, sub_stream_name
...
+        singer.write_bookmark(Context.state,
+                              sub_stream_name,
+                              STREAM_REPLICATION_KEY.get(sub_stream_name),
+                              start_date)
+        Context.state.get("bookmarks").pop(sub_stream_name, None)
+
+    singer.write_state(Context.state)
+
+
+def sync(logger=LOGGER):
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/stripe/__init__.py:1177]: This method is not used here

File: mage_integrations/mage_integrations/sources/stripe/__init__.py (hunk 2 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/stripe/__init__.py]: Modified on 028336556fca308242837d69d26186c17481362d

File: mage_integrations/mage_integrations/sources/stripe/__init__.py (hunk 4 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/stripe/__init__.py]: Modified on 028336556fca308242837d69d26186c17481362d

File: mage_integrations/mage_integrations/sources/stripe/__init__.py (hunk 5 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/stripe/__init__.py]: Modified on 028336556fca308242837d69d26186c17481362d

File: mage_integrations/mage_integrations/sources/stripe/__init__.py (hunk 6 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/stripe/__init__.py]: Removed on 028336556fca308242837d69d26186c17481362d

File: mage_integrations/mage_integrations/sources/stripe/__init__.py (hunk 1 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/stripe/__init__.py:210]: Added in 028336556fca308242837d69d26186c17481362d

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**